build/syntax-highlight.js

```diff
@@ -68,24 +68,113 @@ function syntaxHighlight($, doc) {
     if (!match) {
       return;
     }
-    const name = ALIASES.get(match[1]) || match[1];
-    if (IGNORE.has(name)) {
+    const language = ALIASES.get(match[1]) || match[1];
+    if (IGNORE.has(language)) {
       // Seems to exist a couple of these in our docs. Just bail.
       return;
     }
-    const grammar = Prism.languages[name];
+    const grammar = Prism.languages[language];
     if (!grammar) {
       console.warn(
-        `Unable to find a Prism grammar for '${name}' found in ${doc.mdn_url}`
+        `Unable to find a Prism grammar for '${language}' found in ${doc.mdn_url}`
       );
       return; // bail!
     }
+
+    const addLineNumbers = className.match(/linenumbers/);
     const code = $pre.text();
-    const html = Prism.highlight(code, grammar, name);
-    const $code = $("<code>").html(html);
+    let html = "";
+
+    if (!addLineNumbers) {
+      html = Prism.highlight(code, grammar, language);
+    } else {
+      let highlights = className.match(/(?<=-)\d+/g);
+      highlights = highlights ? highlights : [];
```

Comment:
I've committed the changes. Also added warning messages as suggested by @teoli2003.

> Did you see [prismjs.com/plugins/line-numbers](https://prismjs.com/plugins/line-numbers/) by any chance and try out if that (simpler solution) would work for us at all?

Good question! That was the first thing I did when I started the work. We can't use the plugin because it works only in browser environments. And we use Prism only on server side.